recommender/engine/rl_agent/models/actor_model.py

```diff
@@ -1,23 +1,56 @@
+# pylint: disable=missing-module-docstring, invalid-name, too-many-arguments, no-member
+
+from typing import Tuple
+
 import torch
+from torch import nn
+import torch.nn.functional as F
+
+from recommender.engine.rl_agent.models.history_embedder import HistoryEmbedder
+
 
+class ActorModel(nn.Module):
+    """Actor neural network representing a deterministic policy used by RLAgent"""
 
-class ActorModel(torch.nn.Module):
-    def __init__(self, K: int, SE: int, history_embedder: torch.nn.Module):
+    def __init__(
+        self,
+        K: int,
+        SE: int,
+        UE: int,
+        FE: int,
+        SPE: int,
+        history_embedder: HistoryEmbedder,
+        layer_sizes: Tuple[int] = (256, 512, 256),
+    ):
+        """
+        Args:
+            K: number of services to recommend
+            SE: service embedding dimension
+            UE: user embedding dimension
+            FE: filter embedding dimension
+            SPE search phrase embedding dimension
+            history_embedder: pytorch module implementing history embedding
+            layer_sizes: list containing number of neurons in each hidden layer
+        """
         super().__init__()
         self.K = K
         self.SE = SE
         self.history_embedder = history_embedder
-        # WARNING: history_embedder is a model shared between actor and critic,
-        # the .detach will be probably needed for proper training
 
-        # TODO: layers initialization
-        self.output_layer = torch.nn.Linear(None, K*SE)
+        self.layers = [nn.Linear(SE + UE + FE + SPE, layer_sizes[0])]
+        self.layers += [
+            nn.Linear(n_size, next_n_size)
+            for n_size, next_n_size in zip(layer_sizes, layer_sizes[1:])
+        ]
+        self.layers += [(nn.Linear(layer_sizes[-1], K * SE))]
```

Comment:
If you want to have layers in the list inside the cutome pytorch module you should use [ModuleList](https://pytorch.org/docs/stable/generated/torch.nn.ModuleList.html) or [Sequential](https://pytorch.org/docs/stable/generated/torch.nn.Sequential.html) (and pass *layers_list to it). In other case parameters (weights) of the models/layers inside self.layers list will NOT be automatically registered for the ActorModel(torch.nn.Module). As a result actor_model.parameters() will not return all submodules parameters and the model will not learn correctly.